dynamiqs/time_array.py

```diff
@@ -483,45 +426,59 @@ def __init__(
         self.static = jnp.zeros_like(self.arrays[0]) if static is None else static
 
     @property
-    def dtype(self) -> jnp.dtype:
+    def dtype(self) -> np.dtype:
         return self.arrays.dtype
 
     @property
-    def shape(self) -> Tuple[int, ...]:
-        return jnp.Size((*self.factors[0].shape, self.n, self.n))  # (..., n, n)
+    def shape(self) -> tuple[int, ...]:
+        return tuple(self.factors[0].shape, self.n, self.n)  # (..., n, n)
 
     def __call__(self, t: float) -> Array:
-        values = jnp.stack([x(t) for x in self.factors], dim=-1)  # (..., nf)
+        values = jnp.stack([x(t) for x in self.factors], axis=-1)  # (..., nf)
         values = values.reshape(*values.shape, 1, 1)  # (..., nf, n, n)
         return (values * self.arrays).sum(-3) + self.static  # (..., n, n)
```

Comment:
Since now we have a jit, maybe we could do:
```python
if self.static is None:
  return (values * self.arrays).sum(-3)
else:
  return (values * self.arrays).sum(-3) + self.static
```
to avoid an additional sum with a zero-valued tensor. Unless the typing of `self.static` is not handled by the jit.

Reply:
Actually may not be so smart to add if statements for the jit.

Reply:
I'm guessing the jit optimisation ignores addition with a all-0 array as part of the optimizations they do.